tools/scripts/generic.mk

```diff
@@ -333,6 +333,9 @@ ifneq ($(strip $(LSCRIPT)),)
 LSCRIPT_FLAG = -T$(LSCRIPT)
 endif
 
+PHONY += pre_build
+pre_build:
```

Comment:
This should not be an empty rule, please move the relevant changes from commit 3 into this commit.
For example this:
https://github.com/analogdevicesinc/no-OS/pull/1665/commits/a1597cba79c35f19d2b3f69e9098e790abcedfda#diff-9999da706b3b7d7cadd7d2620e3a2a9d9f252c6772b1918d36a74251fb78a4c4R361

Reply:
okay. I will move relevant changes from commit 3 to commit 2.